packages/flutter/lib/src/painting/circle_border.dart

```diff
@@ -18,16 +18,31 @@ import 'edge_insets.dart';
 /// When applied to a rectangular space, the border paints in the center of the
 /// rectangle.
 ///
+/// The [eccentricity] parameter describes how much a circle will deform to
+/// fit the rectangle it is a border for. A value of zero implies no
+/// deformation (a circle touching at least two sides of the rectangle), a
+/// value of one implies full deformation (an oval touching all sides of the
+/// rectangle).
+///
 /// See also:
 ///
+///  * [OvalBorder], which draws a Circle touching all the edges of the box.
 ///  * [BorderSide], which is used to describe each side of the box.
-///  * [Border], which, when used with [BoxDecoration], can also
-///    describe a circle.
+///  * [Border], which, when used with [BoxDecoration], can also describe a circle.
 class CircleBorder extends OutlinedBorder {
   /// Create a circle border.
   ///
   /// The [side] argument must not be null.
-  const CircleBorder({ super.side }) : assert(side != null);
+  const CircleBorder({ super.side, this.eccentricity = 0 })
+      : assert(side != null),
+        assert(eccentricity >= 0.0, 'The eccentricity argument $eccentricity is not greater than or equal to zero.'),
+        assert(eccentricity <= 1.0, 'The eccentricity argument $eccentricity is not less than or equal to one.');
+
+  /// Defines the ratio (0.0-1.0) from which the border will be drawn
+  /// to the longest side of a rectangular box, touching all the sides.
+  /// When 0.0, it draws a circle. When 1.0, it draws an oval.
+  /// This property is ignored when applied to a squared box.
```

Comment:
Can you use the same wording as above?  "a squared box" doesn't make much sense to me.